src/s3_fetchers.rs

```diff
@@ -1,84 +1,102 @@
-use aws_sdk_s3::{Client, Error};
+use aws_sdk_s3::Client;
+use futures::stream::StreamExt;
 
 /// Queries the list of the objects in the bucket, grouped by "/" delimiter.
 /// Returns the continuation token along with the so called list of folder names
 /// that represent a block heights
 pub(crate) async fn list_blocks(
-    client: &Client,
-    bucket: &str,
-    start_after_block_height: Option<String>,
-    continuation_token: Option<String>,
-) -> Result<crate::types::ListObjectResponse, Error> {
-    let response = client
+    s3_client: &Client,
+    s3_bucket_name: &str,
+    start_from_block_height: String,
+) -> anyhow::Result<Vec<String>> {
+    let response = s3_client
         .list_objects_v2()
-        .max_keys(10)
+        .max_keys(1000)
         .delimiter("/".to_string())
-        .set_start_after(start_after_block_height)
-        .set_continuation_token(continuation_token)
-        .bucket(bucket)
+        .start_after(start_from_block_height)
+        .bucket(s3_bucket_name)
         .send()
         .await?;
 
-    let continuation_token = response.next_continuation_token().map(ToOwned::to_owned);
-    let folder_names = match response.common_prefixes() {
+    Ok(match response.common_prefixes() {
         None => vec![],
         Some(common_prefixes) => common_prefixes
-            .into_iter()
+            .iter()
             .filter_map(|common_prefix| common_prefix.prefix.clone())
```

Comment:
Can we use `.into_iter()` and drop the `.clone()`?

Reply:
Nah, "cannot move out common_prefix.prefix which is behind a shared reference"

Reply:
According to the [implementation](https://docs.rs/aws-sdk-s3/latest/src/aws_sdk_s3/output.rs.html#238), instead of using a method `common_prefixes()`, you should be able to use the field `common_prefixes`, which is an `Option<Vec<...>>`.